Packs/MicrosoftExchangeOnPremise/Integrations/EWSv2/EWSv2.py

```diff
@@ -805,8 +811,26 @@ def get_searchable_mailboxes(protocol):  # pragma: no cover
 
 
 def search_mailboxes(protocol, filter, limit=100, mailbox_search_scope=None, email_addresses=None):  # pragma: no cover
+    """
+    Search mailboxes for items matching the given filter.
+
+    Args:
+        protocol (Protocol): The EWS protocol object.
+        filter (str): The search filter to apply.
+        limit (int): The maximum number of results to return. Default value is 100.
+        mailbox_search_scope (str or list, optional): The mailbox search scope. Defaults to None.
+        email_addresses (str, optional): Comma-separated list of email addresses to search. Defaults to None.
+
+    Returns:
+        dict: A dictionary containing the search results.
+
+    Raises:
+        Exception: If both mailbox_search_scope and email_addresses are provided, or if no searchable mailboxes are found.
+    """
     mailbox_ids = []
-    limit = int(limit)
+    limit_argument = arg_to_number(limit)
+    if not limit_argument:
+        raise (DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer."))
     if mailbox_search_scope is not None and email_addresses is not None:
         raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```

Comment:
```suggestion
    if not limit_argument:
        raise (DemistoException(f"Invalid limit value: {limit}. Please provide a valid integer."))
        
    if mailbox_search_scope is not None and email_addresses is not None:
        raise Exception("Use one of the arguments - mailbox-search-scope or email-addresses, not both")
```

Reply:
@omerKarkKatz I removed the redundant brackets around the `DemistoException` if that was this comment's intent, if there is something else, let me know :)